tools/stats/print_test_stats.py

```diff
@@ -693,13 +693,13 @@ def parse_reports(folder: str) -> Dict[str, TestFile]:
 
 def build_info() -> ReportMetaMeta:
     return {
-        "build_pr": os.environ.get("CIRCLE_PR_NUMBER", ""),
-        "build_tag": os.environ.get("CIRCLE_TAG", ""),
-        "build_sha1": os.environ.get("CIRCLE_SHA1", ""),
-        "build_base_commit": get_base_commit(os.environ.get("CIRCLE_SHA1", "HEAD")),
-        "build_branch": os.environ.get("CIRCLE_BRANCH", ""),
+        "build_pr": os.environ.get("PR_NUMBER", ""),
```

Comment:
There are a few more places that need adjusting, you can find them with a search over the codebase:

```bash
# look up all the relevant envs, ignore usage in .circleci files
git grep -E '(CIRCLE_BRANCH)|(CIRCLE_PR_NUMBER)|(CIRCLE_SHA1)|(CIRCLE_TAG)|(CIRCLE_WORKFLOW_ID)' | awk '{print $1}' | sort | uniq | grep -v '\.circleci'
```

Out of these I see these

```
.jenkins/pytorch/test.sh:PR_NUMBER=${PR_NUMBER:-${CIRCLE_PR_NUMBER:-}}
tools/stats/print_test_stats.py:
tools/stats/scribe.py:
tools/stats/upload_binary_size_to_scuba.py:
tools/testing/test_selections.py:
```

Most of these we only use in GitHub Actions, but `test_selections.py` is used in both CircleCI and GitHub Actions so it'll need some logic to look for the GHA env variable and fall back to the CircleCI env variable if it's not found (i.e. what `test.sh` with the line `sh:PR_NUMBER=${PR_NUMBER:-${CIRCLE_PR_NUMBER:-}}`)

Reply:
Sounds good @driazati will update the diff accordingly